Game Speed / Effective Multiplier

petethered opened the pull request:
This adds "Game Speed" to the Run reports

<img width="500" height="164" alt="image" src="https://github.com/user-attachments/assets/97a40820-2627-4a30-ac2a-9c6070b3131b" />

It's force calculated as 3 digits of significance (1.234) , displayed with a trailing x (1.234x) 

It's added as a field, so it also graphs.

<img width="551" height="240" alt="image" src="https://github.com/user-attachments/assets/42f976f4-92d0-4824-82d6-14ca7efb647b" />

The graphs don't QUITE make sense once you move out of the "per run", but neither do the game time / run time graphs so seems fine.

---

code-organization-naming: 

*No changes required - organization and naming are well-structured.*

---

frontend-design-reviewer:

As per discord, it changed it to remove trailing zeros.

Commit:
Add Game Speed to the run report

Commit:
Adds in base changes to make game speed a graphable field

Commit:
All the test code

Commit:
Design agent changes for removing of trailing zeros

petethered commented:
Side note:

"Game Speed" (and gameSpeed) were used as labels/variables due to the language in the perk

Increase Max Game Speed by +1.00

So it adheres to goal of keeping naming understandable and consistent

jarekb84 commented on src/shared/formatting/run-display-formatters.ts:
```diff
@@ -64,3 +64,19 @@ export function formatDurationHoursMinutes(durationSeconds: number): string {
+ * Examples: 2.0 -> "2x", 2.5 -> "2.5x", 2.123 -> "2.123x"
+ * Game speed is the ratio of gameTime to realTime.
+ */
+export function formatGameSpeed(value: number): string {
```
As per the discord convo, lets make this a lil more generic to handle multipliers.

jarekb84 commented on src/features/analysis/shared/parsing/data-parser.ts:
```diff
@@ -136,13 +137,18 @@ function extractKeyStatsFromFields(fields: Record<string, GameRunField>): {
+
+  const realTime = (fields.realTime?.value as number) || 0;
+  const gameTime = (fields.gameTime?.value as number) || 0;
+  const gameSpeed = realTime === 0 ? null : Math.round((gameTime / realTime) * 1000) / 1000;
```
Extract a common function to calculate this and use it in this file and in csv-field-mapping.ts since it seems this is exactly the same calc, and there is quite a bit of conditional logic here.

petethered committed:
Merge branch 'jarekb84:main' into effective_mult

jarekb84 committed:
FEATURE: Add Activity Heatmap for weekly run coverage visualization (#191) (#192)

Introduce a new Activity Heatmap analysis tab that displays a 24x7
(hours x days) grid showing when game runs occurred during a given week.
Players can visualize run coverage patterns, identify idle time gaps,
and optimize their farming strategies.

Key components:
- Core data engine: grid builder, week utilities, run time distribution
- Week navigation with prev/next/current controls
- Tier and run type filtering with active hours configuration
- Summary statistics: daily/weekly coverage, run type breakdown, idle time
- Cell tooltips with per-segment detail and gap analysis
- localStorage persistence for filter/active-hours preferences
- Comprehensive unit test suite across all pure logic modules

Architecture: Sunday-through-Saturday week model with locale-aware
day/date header labels, DOM-based segment rendering (replacing CSS
gradient approach), and three-layer separation (logic/hooks/components).

petethered committed:
Merge branch 'jarekb84:main' into effective_mult

Commit:
Updates formats
